model/qti/container/Container.php

```diff
@@ -366,4 +372,46 @@ public function toArray($filterVariableContent = false, &$filtered = array()){
         return $data;
     }
 
+    /**
+     * Remove xhtml elements which have namespace example. <ht5:ruby>element</ht5>.
+     *
+     * @param $html
+     * @param array $namespaces
+     * @return null|string|string[]
+     */
+    protected function stripNamespaces($html, $namespaces)
+    {
+        $regexArray = [];
+
+        foreach ($namespaces as $namespace) {
+            $regexArray[] = "/[$namespace]+:([$namespace]+[=>])/";
```

Comment:
I am not sure about the resulting regex. I have the feeling it is bugprone.

For instance, if the `$namespace` is `"ht5"`, as in your sample, we will get this regex: `/[ht5]+:([ht5]+[=>])/`
As per the `preg_replace` below, you will replace the matched string by the first expression, that is the namespace itself (and only string like `"ht5:ht5>"` will be mathed).
Moreover, your regex will match any strings like: `"hhhh:555="` or `"55hhtthh:tt>"` , but not `ht5:ruby>`.

A better regex would be: `'/' . preg_quote($namespace) . ':([^=>]+?[=>])/'`

And there is also no protection against possible unauthorized chars.

Reply:
I did test this regex with <qt5:ruby>test</qt5:ruby> and it was working fine.
I will give a try with your example also, and let you know

Reply:
Well did a couple of tests more, striping the namespace here it's problematic because this method it's also called when we import a file, So we are going to have the xml imported with namespace stripped. I think we will have to strip namespaces in on the frontend side, much safer. Can you offer some guidance where I can do this?

Reply:
I am not an expert in this area. Maybe @ssipasseuth could help you more than me.
However, for the regex I can help you as I have a good experience in this domain.